exercises-cpp/mohammed_almakki/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,123 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <regex>
+#include <unordered_map>
+#include <cstring>
+#include <iomanip>
+
+std::string readFile(const std::string &filepath)
+{
+    std::ifstream inputStream(filepath);
+
+    if (inputStream)
+    {
+        std::ostringstream ss;
+        ss << inputStream.rdbuf();
+        return ss.str();
+    }
+    else
+    {
+        throw std::invalid_argument("Invalid path. Please make sure the file path is correct and the file exists.");
+    }
+}
+
+std::vector<std::string> tokenizeFile(const std::string &file)
+{
+    std::regex re("[-\\s]");
+    std::sregex_token_iterator first{file.begin(), file.end(), re, -1}, last;
+    std::vector<std::string> tokens{first, last};
+    return tokens;
+}
+
+std::unordered_map<std::string, uint32_t> countWords(const std::vector<std::string> words)
+{
+    std::unordered_map<std::string, uint32_t> wordsCounts;
+    for (auto word : words)
+    {
+        std::transform(
+            word.begin(), word.end(), word.begin(),
+            [](unsigned char c)
+            { return std::tolower(c); });
+        word.erase(std::remove_if(word.begin(), word.end(), ispunct), word.end());
+        if (word.length() > 4)
+        {
+            wordsCounts[word]++;
```

Comment:
you should have a check to see if word is currently in the wordCounts map

Reply:
Hmmm, why is it needed?
If there is a get operation, then I need a check, but this is just used as a counter.

Reply:
Ultimatly it is not required for it to work, but it is better practice to have the check when performing an action on an object in a map where it is not certain that the object will exist.